internal/transport/middleware.go

```diff
@@ -161,6 +161,10 @@ func (m middleware) datadogTracer(next http.Handler) http.Handler {
 			resourceName = "unknown"
 		}
 		resourceName = r.Method + " " + resourceName
+		resourcePrefix := r.Header.Get("datadog-resource-prefix")
+		if resourcePrefix != "" {
```

Comment:
Add a comment on this `if` to explain the meaning of this header (ideally on all of them).